### DISCUSSION TITLE
How to format Date and Time column

### DISCUSSION THREAD

**Discussion by egwrk** [microsoft-search/pnp-modern-search-layouts#4] [Q&A] [answered]
I'm new to HTML handlebars and I'm trying to use it to format the dates in my PnP - Search Results web part but with no luck. By default, the date is displayed like this: 2024-06-19T04:00:00Z

Does anyone know how to format the date so it displays like this instead: 19 June 2024
I'm using the details list layout and also prefer not to use an HTML template.

I've tried using each of the lines below but the date column goes blank:

```
{{getDate HolidayDate "dd MMM yyyy"}}

{{getDate slot item @root.slots.HolidayDate "dd MMM yyyy"}}

{{getDate HolidayDate dateFormat "dd MMM yyyy"}}

{{getDate slot item @root.slots.HolidayDate dateFormat "dd MMM yyyy"}}
```

**Comment by egwrk** [comment, marked as answer]
Found the answer here: https://github.com/microsoft-search/pnp-modern-search/discussions/3468

This is the syntax I used and it worked!
{{getDate item.RefinableDate17 'DD MMMM yyyy' 0 true}}

**Comment by phil-maynard** [comment]
For anyone else having this issue, there's useful guidance on the [PnP Modern Search - Templating page for the **getDate** Handlebar helper](https://microsoft-search.github.io/pnp-modern-search/extensibility/templating/#getdate). It's also worth referencing the [Moment.js documentation](https://momentjs.com/docs/#/parsing/string-format/) that contains information on the string format. I suspect that the key issue in the example given was the use of the double quote character (") to format the date rather then the single quote ('). If this was within a string segment, it likely caused an error which the webpart resolves but not displaying anything.
